### PR TITLE
[Paddle Inference] Add multi_query_attention kernel

### PR DESCRIPTION
**Pull request by superxf**
### PR types

others

### PR changes

OPs

### Description

- Support masked multiquery attention for transformer decoder stage.
- Support dtype：[float | float16 | bfloat16]
- export python api： from paddle.incubate.nn.functional import masked_multiquery_attention

Pcard-74485

### CODE REVIEW

File: python/paddle/incubate/nn/functional/masked_multiquery_attention.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+    Examples:
+        .. code-block:: python
+
+            # required: gpu
+            import paddle
+            import paddle.incubate.nn.functional as F
+
+            # input: [batch_size, num_head, dim_head]
+            q = paddle.rand(shape=(2, 32, 128), dtype="float32")
```
Comment (vivienfanghuagood) [inline python/paddle/incubate/nn/functional/masked_multiquery_attention.py:78]: 例子建议给一个group=1和group=2的例子吧

**Comment by luotao1** [comment]
@superxf 如果是内部同学，可以在 http://rdmp.dltp.baidu-int.com/data-management/github-user 登记下

File: python/paddle/incubate/nn/functional/masked_multiquery_attention.py (hunk 1 above)
Comment (superxf) [reply python/paddle/incubate/nn/functional/masked_multiquery_attention.py:78]: Done

File: paddle/phi/kernels/fusion/gpu/masked_multiquery_attention.h
Comment (xiaoxiaohehe001) [inline paddle/phi/kernels/fusion/gpu/masked_multiquery_attention.h:1]: 可以参考 https://github.com/PaddlePaddle/Paddle/pull/56411 ，把 .h 文件消除掉

**Comment by carryyu** [comment]
谢谢，已收到！
